android/library/src/main/java/com/liferay/mobile/screens/ddl/NumberValidator.kt

```diff
@@ -15,64 +15,72 @@
 package com.liferay.mobile.screens.ddl
 
 import java.text.NumberFormat
-import java.util.regex.Pattern
 
 /**
  * @author Victor Oliveira
  */
-abstract class NumberValidator(val right: Number, val errorMessage: String) {
-	abstract fun validate(left: Number): Boolean
-
-	companion object {
-
-		private val PATTERN: Pattern = Pattern.compile("[^0-9]+([0-9]+)$")
-
-		@JvmStatic
-		fun parseNumberValidator(validationMap: Map<String, Any>): NumberValidator {
-			val errorMessage = validationMap["error"] as? String ?: ""
-			val expression = validationMap["expression"] as? String ?: ""
-
-			val matcher = PATTERN.matcher(expression)
-			if (matcher.find() && matcher.groupCount() >= 1) {
-				val numberStr = matcher.group(1)
-				val operator = matcher.group(0).removeSuffix(numberStr)
-				val number = NumberFormat.getInstance().parse(numberStr)
-
-				return when (operator) {
-					">=" -> IsGreaterThanOrEqualTo(number, errorMessage)
-					">" -> IsGreaterThan(number, errorMessage)
-					"==" -> IsEqualTo(number, errorMessage)
-					"<=" -> IsLessThanOrEqualTo(number, errorMessage)
-					"<" -> IsLessThan(number, errorMessage)
-					else -> Unknown(NumberFormat.getInstance().parse("0"), "Unknown Error")
-				}
-			}
-
-			return Unknown(NumberFormat.getInstance().parse("0"), "Unknown Error")
-		}
-	}
-
-	class IsGreaterThanOrEqualTo(right: Number, errorMessage: String) : NumberValidator(right, errorMessage) {
-		override fun validate(left: Number): Boolean = left.toDouble() >= right.toDouble()
-	}
-
-	class IsGreaterThan(right: Number, errorMessage: String) : NumberValidator(right, errorMessage) {
-		override fun validate(left: Number): Boolean = left.toDouble() > right.toDouble()
-	}
-
-	class IsEqualTo(right: Number, errorMessage: String) : NumberValidator(right, errorMessage) {
-		override fun validate(left: Number): Boolean = left.toDouble() == right.toDouble()
-	}
-
-	class IsLessThanOrEqualTo(right: Number, errorMessage: String) : NumberValidator(right, errorMessage) {
-		override fun validate(left: Number): Boolean = left.toDouble() <= right.toDouble()
-	}
-
-	class IsLessThan(right: Number, errorMessage: String) : NumberValidator(right, errorMessage) {
-		override fun validate(left: Number): Boolean = left.toDouble() < right.toDouble()
-	}
-
-	class Unknown(number: Number, errorMessage: String) : NumberValidator(number, errorMessage) {
-		override fun validate(left: Number): Boolean = true
-	}
+abstract class NumberValidator(val right: Number, val errorMessage: String) : LocalValidator {
+    abstract fun validate(left: Number): Boolean
+    override fun getCustomErrorMessage(): String {
+        return errorMessage
+    }
+
+    companion object {
+
+        private val COMPOUND_OPERATOR_REGEX = Regex("((?=(?!^)<=|==|>=)(?<!<=|==|>=)|(?!<=|==|>=)(?<=<=|==|>=))")
```

Comment:
🤯

Reply:
🤣@victorlaerte can explain to you

Reply:
@victorlaerte proceed xD

Reply:
haha, this regex is to split the string when find <=, ==, >= but keeping the delimiter. There's another one that tests for <,> because with I add this to the same regex it can split compound operators like <= and >= in two (<, = and >, =). I tried to solve this in only one regex and keep delimiters but it was easier to do this way.

Reply:
Got it! thanks!